File: python/atbash-cipher/atbash_cipher.py
from string import ascii_lowercase
import re

cipher = str.maketrans(ascii_lowercase, ascii_lowercase[::-1])

def encode(plain_text):
    ciphered = re.sub("\W", "", plain_text).replace("_", "").lower().translate(cipher)
    return " ".join(ciphered[x:x+5] for x in range(0, len(ciphered), 5))


def decode(ciphered_text):
    return ciphered_text.replace(" ", "").translate(cipher)

File: clojure/atbash-cipher/src/atbash_cipher.clj
(ns atbash-cipher
  (:require [clojure.string :as str]))

(def cipher
  (let [alphabet (map char (range (int \a) (inc (int \z))))]
    (zipmap alphabet (reverse alphabet))))

(defn encode
  [text]
  (let [ciphered (-> text
                     str/lower-case
                     (str/replace #"\W" "")
                     (->> (map (some-fn cipher identity))))]
    (->> ciphered
         (partition-all 5)
         (map #(apply str %))
         (str/join " "))))

(defn decode
  "Decode the space-separated ciphertext into unseparated cleartext. The Python
  version of this problem asks for this, but the Clojure version doesn't, so I
  thought I'd throw it in anyway."
  [text]
  (-> text
      (str/replace " " "")
      (->> (map (some-fn cipher identity))
           (str/join ""))))
